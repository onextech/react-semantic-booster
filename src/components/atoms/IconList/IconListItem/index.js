import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Header } from 'semantic-ui-react';
import styled from 'styled-components';
import { getCustomClassName, subtractObject } from '../../../../utils/helpers';


const verticalClassName = 'vertical';

const StyledIconListItem = styled.div`
  display: flex;
  .icon { margin-right: .3em; }
  .text { margin-top: .15em; }
  h5.ui.header {
    font-size: 1.2em;
    margin-bottom: 0;
  }
  &.${verticalClassName} {
    flex-direction: column;
    text-align: center;
    .icon { margin-right: 0; }
    .text { margin-top: .6em; } 
  }
`;

const IconListItem = (props) => {
  const { icon, content, header } = props;

  // 1. Define custom props for this component
  const classProps = {
    vertical: verticalClassName,
  };
  // 1.5. Define default props
  const defaultProps = {
    vertical: false,
  };
  const customProps = {};

  const allProps = { ...defaultProps, ...props };
  // 2. Render the custom class names
  const className = props &&
    getCustomClassName(classProps, allProps);
  // 3. Clean up custom props from main prop set to avoid prop clashing
  const semanticProps = props &&
    subtractObject({ ...defaultProps, ...classProps }, allProps);

  return (
    <StyledIconListItem {...semanticProps} {...customProps} className={className}>
      <div className="icon">
        <Icon name="check circle" size="large" {...icon} />
      </div>
      <div className="text">
        {header && <Header as="h5">{header}</Header>}
        {content && <p>{content}</p>}
      </div>
    </StyledIconListItem>
  );
};

IconListItem.propTypes = {
  icon: PropTypes.object,
  content: PropTypes.string,
  header: PropTypes.string,
  vertical: PropTypes.bool,
  className: PropTypes.string,
};

export default IconListItem;
