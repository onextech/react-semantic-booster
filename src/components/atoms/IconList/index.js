import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { getCustomClassName, subtractObject } from '../../../utils/helpers';
import IconListItem from './IconListItem';


const iconListItemClassName = 'icon-list-item';
const verticalClassName = 'vertical';
const centerClassName = 'center';

const StyledIconList = styled.div`
  .${iconListItemClassName} {
    &:not(:last-child) { margin-bottom: 1em; }
  }
  &.${centerClassName} {
    .${iconListItemClassName} {
      justify-content: center;
    }
  }
  ${({ textAlign }) => textAlign && `text-align: ${textAlign};`}
  `;

const IconList = (props) => {
  const {
    items, vertical, textAlign, headerProps, contentProps, icon,
  } = props;

  const iconListItemProps = {
    vertical, headerProps, contentProps, icon,
  };

  // 1. Define custom props for this component
  const classProps = {
    vertical: verticalClassName,
    center: centerClassName,
  };
  // 1.5. Define default props
  const defaultProps = {
    vertical: true,
    center: false,
  };
  const customProps = {
    textAlign,
  };

  const allProps = { ...defaultProps, ...props };
  // 2. Render the custom class names
  const className = props &&
    getCustomClassName(classProps, allProps);
  // 3. Clean up custom props from main prop set to avoid prop clashing
  const semanticProps = props &&
    subtractObject({ ...defaultProps, ...classProps }, allProps);

  return (
    <StyledIconList {...semanticProps} {...customProps} className={className}>
      {
        items.map((item, i) => (
          <IconListItem
            key={i}
            {...iconListItemProps}
            {...item}
            className={iconListItemClassName} />
        ))
      }
    </StyledIconList>
  );
};

IconList.propTypes = {
  items: PropTypes.array.isRequired,
  vertical: PropTypes.bool,
  center: PropTypes.bool,
  textAlign: PropTypes.oneOf(['left', 'right', 'center']),
  headerProps: PropTypes.object,
  contentProps: PropTypes.object,
  icon: PropTypes.object,
};

IconList.Item = IconListItem;

export default IconList;
