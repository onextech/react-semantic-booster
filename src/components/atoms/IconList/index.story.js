import React from 'react';
import { storiesOf } from '@storybook/react';
import { Grid } from 'semantic-ui-react';
import Block from '../Block';
import IconList from '.';


const exampleProps = {
  header: 'Hello World',
  content: 'The property will be presented to over 120 agents during the Rodeo Realty office meeting',
};

// eslint-disable-next-line import/prefer-default-export
export const items = [exampleProps, exampleProps, exampleProps];

storiesOf('IconList', module)
  .add('Default', () => (
    <div>
      <Block>
        <Grid container stackable>
          <Grid.Row columns="equal">
            <Grid.Column>
              <IconList.Item {...exampleProps} vertical />
            </Grid.Column>
            <Grid.Column>
              <IconList.Item {...exampleProps} vertical />
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Block>
      <Block>
        <IconList center items={items} />
      </Block>
      <Block inverted>
        <IconList
          icon={{ name: 'star' }}
          headerProps={{ inverted: true }}
          center
          items={items}
        />
      </Block>
    </div>
  ));
