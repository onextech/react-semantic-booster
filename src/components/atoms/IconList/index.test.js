import React from 'react';
import renderer from 'react-test-renderer';
import IconList from '.';
import { items } from './index.story';


describe('<IconList>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<IconList items={items} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
