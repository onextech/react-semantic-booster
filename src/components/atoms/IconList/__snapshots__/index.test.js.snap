// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconList> renders correctly 1`] = `
<div
  className="vertical  sc-bwzfXH lkruZO"
>
  <div
    className="   icon-list-item sc-bdVaJa iyPRyX"
    content="The property will be presented to over 120 agents during the Rodeo Realty office meeting"
  >
    <div
      className="icon"
    >
      <i
        aria-hidden="true"
        className="check circle large icon"
      />
    </div>
    <div
      className="text"
    >
      <h5
        className="ui header"
      >
        Hello World
      </h5>
      <p>
        The property will be presented to over 120 agents during the Rodeo Realty office meeting
      </p>
    </div>
  </div>
  <div
    className="   icon-list-item sc-bdVaJa iyPRyX"
    content="The property will be presented to over 120 agents during the Rodeo Realty office meeting"
  >
    <div
      className="icon"
    >
      <i
        aria-hidden="true"
        className="check circle large icon"
      />
    </div>
    <div
      className="text"
    >
      <h5
        className="ui header"
      >
        Hello World
      </h5>
      <p>
        The property will be presented to over 120 agents during the Rodeo Realty office meeting
      </p>
    </div>
  </div>
  <div
    className="   icon-list-item sc-bdVaJa iyPRyX"
    content="The property will be presented to over 120 agents during the Rodeo Realty office meeting"
  >
    <div
      className="icon"
    >
      <i
        aria-hidden="true"
        className="check circle large icon"
      />
    </div>
    <div
      className="text"
    >
      <h5
        className="ui header"
      >
        Hello World
      </h5>
      <p>
        The property will be presented to over 120 agents during the Rodeo Realty office meeting
      </p>
    </div>
  </div>
</div>
`;
